src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -730,45 +748,16 @@ void InitializeTabularMonthly(EnergyPlusData &state)
             AvgSumVar = ort->MonthlyFieldSetInput(FirstColumn + colNum - 1).varAvgSum;
             StepTypeVar = ort->MonthlyFieldSetInput(FirstColumn + colNum - 1).varStepType;
             UnitsVar = ort->MonthlyFieldSetInput(FirstColumn + colNum - 1).varUnits;
-            //    DO iKey = 1, KeyCount  !noel
-            //       NamesOfKeys(iKey) = MonthlyFieldSetInput(FirstColumn + ColNum - 1)%NamesOfKeys(iKey)  !noel
-            //       IndexesForKeyVar(iKey) = MonthlyFieldSetInput(FirstColumn + ColNum - 1)%IndexesForKeyVar(iKey) !noel
-            //    ENDDO
-            // #else
-            //    curVariMeter = UtilityRoutines::makeUPPER(MonthlyFieldSetInput(FirstColumn + ColNum - 1)%variMeter)
-            //    ! call the key count function but only need count during this pass
-            //    CALL GetVariableKeyCountandType(state, curVariMeter,KeyCount,TypeVar,AvgSumVar,StepTypeVar,UnitsVar)
-            //    ALLOCATE(NamesOfKeys(KeyCount))
-            //    ALLOCATE(IndexesForKeyVar(KeyCount))
-            //    CALL GetVariableKeys(state, curVariMeter,TypeVar,NamesOfKeys,IndexesForKeyVar)
-            // #endif
 
-            if (KeyCount == 0) {
-                ++state.dataOutRptTab->ErrCount1;
-                if (state.dataOutRptTab->ErrCount1 == 1 && !state.dataGlobal->DisplayExtraWarnings &&
-                    state.dataGlobal->KindOfSim == Constant::KindOfSim::RunPeriodWeather) {
-                    ShowWarningError(state, "Processing Monthly Tabular Reports: Variable names not valid for this simulation");
-                    ShowContinueError(state, "...use Output:Diagnostics,DisplayExtraWarnings; to show more details on individual variables.");
-                }
-                // fixing CR5878 removed the showing of the warning once about a specific variable.
-                if (state.dataGlobal->DisplayExtraWarnings && state.dataGlobal->KindOfSim == Constant::KindOfSim::RunPeriodWeather) {
-                    ShowWarningError(state, format("Processing Monthly Tabular Reports: {}", ort->MonthlyInput(TabNum).name));
-                    ShowContinueError(state, format("..Variable name={} not valid for this simulation.", curVariMeter));
-                    if (state.dataOutRptTab->VarWarning) {
-                        ShowContinueError(state,
-                                          "..Variables not valid for this simulation will have \"[Invalid/Undefined]\" in the Units Column of "
-                                          "the Table Report.");
-                        state.dataOutRptTab->VarWarning = false;
-                    }
-                }
+            if (KeyCount == 0 && issueWarnings && !ort->MonthlyInput(TabNum).isNamedMonthly) {
+                ShowWarningError(
+                    state, format("In Output:Table:Monthly '{}' invalid Variable or Meter Name '{}'", ort->MonthlyInput(TabNum).name, curVariMeter));
```

Comment:
Print the detail only if it's not a namedMonthlyOne and issueWArnings (= DisplayExtraWarnings + a Weather file run requested)